frontend/lib/src/SessionInfo.ts

```diff
@@ -49,6 +49,20 @@ export class SessionInfo {
    */
   private _last?: Props
 
+  /**
+   * An ID uniquely identifying the last ForwardMsg that we received from the
+   * server.
+   */
+  private _lastForwardMsgId?: string
+
+  // FIXME(vdonato): Either switch to client-tracked sequence numbers or get
```

Comment:
> Either switch to client-tracked sequence numbers
isn't `_forwardMsgSequenceNumber` tracked by the client?

Reply:
Yeah, although whether or not we want to keep things that way + whether we're able to guarantee that no race conditions are possible with this approach is why there's a FIXME here. For now, I just threw a solution together since it's easier to test if we're tracking sequence numbers on the client (as no GS/python side changes need to be deployed).